src/libraries/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -95,40 +94,67 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
+            ThrowIfDisposed();
             ValidateBufferArguments(buffer, offset, count);
-            _buffer.Write(buffer, offset, count);
+            _internalStream.Write(new(buffer, offset, count));
+            Flush();
         }
 
-        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            ThrowIfDisposed();
             ValidateBufferArguments(buffer, offset, count);
-            return _buffer.WriteAsync(buffer, offset, count, cancellationToken);
+            await _internalStream.WriteAsync(buffer.AsMemory(offset, count), cancellationToken).ConfigureAwait(false);
+            await FlushAsync(cancellationToken).ConfigureAwait(false);
         }
 
-        public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+        public override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
         {
-            return _buffer.WriteAsync(buffer, cancellationToken);
+            ThrowIfDisposed();
+            await _internalStream.WriteAsync(buffer, cancellationToken).ConfigureAwait(false);
+            await FlushAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Why are we injecting flushes on writes?

Reply:
To eliminate mini write buffer on HttpConnection. (Every write will be different chunk)

Reply:
Small buffers like this are fine though, no? The user is free to call `Flush` when they want to ensure data is written.
Manually flushing here only guarantees that each user `Write` call will translate into a `Write` call to the transport stream (e.g. Socket). The HTTP traffic sent will be the same.

Reply:
It is, I didn't realize HTTP traffic will not change, once I looked at the code path now I got it, I can remove flushes